mlflow/metrics/metric_definitions.py

```diff
@@ -507,3 +507,32 @@ def _fn(predictions, targets):
         return MetricValue(scores=scores, aggregate_results=standard_aggregations(scores))
 
     return _fn
+
+
+def _bleu_eval_fn(predictions, targets=None, metrics=None):
+    if not _validate_text_data(targets, "bleu", targets_col_specifier) or not _validate_text_data(
+        predictions, "bleu", predictions_col_specifier
+    ):
+        return
+
+    try:
+        bleu = _cached_evaluate_load("bleu")
+    except Exception as e:
+        _logger.warning(f"Failed to load 'bleu' metric (error: {e!r}), skipping metric logging.")
+        return
+
+    result = []
+    for i, (prediction, target) in enumerate(zip(predictions, targets)):
+        if len(target) == 0:
```

Comment:
Do we also want to check if the prediction length is 0?

Reply:
Good point! Added a check for empty `predictions`, similar to `targets`, with a warning and a placeholder value `0`.